integrationTests/testInitializer.go

```diff
@@ -0,0 +1,1931 @@
+func CreateTrieStorageManagerWithPruningStorer(coordinator sharding.Coordinator, notifier pruning.EpochStartNotifier) common.StorageManager {
+	generalCfg := config.TrieStorageManagerConfig{
+		PruningBufferLen:      1000,
+		SnapshotsBufferLen:    10,
+		SnapshotsGoroutineNum: 1,
+	}
+
+	mainStorer, _, err := testStorage.CreateTestingTriePruningStorer(coordinator, notifier)
+	if err != nil {
+		fmt.Println("err creating main storer" + err.Error())
+	}
+	checkpointsStorer, _, err := testStorage.CreateTestingTriePruningStorer(coordinator, notifier)
+	if err != nil {
+		fmt.Println("err creating checkpoints storer" + err.Error())
+	}
+	args := trie.NewTrieStorageManagerArgs{
+		MainStorer:             mainStorer,
+		CheckpointsStorer:      checkpointsStorer,
+		Marshalizer:            TestMarshalizer,
+		Hasher:                 TestHasher,
+		GeneralConfig:          generalCfg,
+		CheckpointHashesHolder: hashesHolder.NewCheckpointHashesHolder(10000000, uint64(TestHasher.Size())),
+		IdleProvider:           &testscommon.ProcessStatusHandlerStub{},
+	}
+	trieStorageManager, _ := trie.NewTrieStorageManager(args)
+
+	return trieStorageManager
+}
+
+// CreateTrieStorageManager creates the trie storage manager for the tests
+func CreateTrieStorageManager(store storage.Storer) (common.StorageManager, storage.Storer) {
+	generalCfg := config.TrieStorageManagerConfig{
+		PruningBufferLen:      1000,
+		SnapshotsBufferLen:    10,
+		SnapshotsGoroutineNum: 1,
+	}
+	args := trie.NewTrieStorageManagerArgs{
+		MainStorer:             store,
+		CheckpointsStorer:      CreateMemUnit(),
+		Marshalizer:            TestMarshalizer,
+		Hasher:                 TestHasher,
+		GeneralConfig:          generalCfg,
+		CheckpointHashesHolder: hashesHolder.NewCheckpointHashesHolder(10000000, uint64(TestHasher.Size())),
+		IdleProvider:           &testscommon.ProcessStatusHandlerStub{},
+	}
+	trieStorageManager, _ := trie.NewTrieStorageManager(args)
+
+	return trieStorageManager, store
+}
+
+// CreateAccountsDB creates an account state with a valid trie implementation but with a memory storage
+func CreateAccountsDB(
+	accountType Type,
+	trieStorageManager common.StorageManager,
+) (*state.AccountsDB, common.Trie) {
+	tr, _ := trie.NewTrie(trieStorageManager, TestMarshalizer, TestHasher, maxTrieLevelInMemory)
+
+	ewl, _ := evictionWaitingList.NewEvictionWaitingList(100, memorydb.New(), TestMarshalizer)
+	accountFactory := getAccountFactory(accountType)
+	spm, _ := storagePruningManager.NewStoragePruningManager(ewl, 10)
+	args := state.ArgsAccountsDB{
+		Trie:                  tr,
+		Hasher:                sha256.NewSha256(),
+		Marshaller:            TestMarshalizer,
+		AccountFactory:        accountFactory,
+		StoragePruningManager: spm,
+		ProcessingMode:        common.Normal,
+		ProcessStatusHandler:  &testscommon.ProcessStatusHandlerStub{},
+	}
+	adb, _ := state.NewAccountsDB(args)
+
+	return adb, tr
+}
+
+func getAccountFactory(accountType Type) state.AccountFactory {
```

Comment:
we definitely do not need this :D

Reply:
I think we need. We should still have validator statistics .